### PR TITLE
Adds a new smite that slams the target across many tables on the station

### PR DESCRIPTION
**Pull request by DuffCreeper**
## About The Pull Request

Basically when the target is hit with this smite, they get table slammed for about 1 minute and 30 seconds across basically every table in the station. End result is probably a 550 brute body. Also generates a CC report informing the station that this person probably did something bad to deserve it

## Why It's Good For The Game

https://cdn.discordapp.com/attachments/758141117987422239/844935097408159804/moth_abuse.mp4
When your 27mb file too big for github to allow you to upload 👌
Also volume warning since OBS made it louder than it should

## Changelog
:cl:
add: Added a new smite to allow for endless table slamming fun
/:cl:

### CODE REVIEW

File: code/modules/admin/smites/tabletidestationwide.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	priority_announce("[target] has brought the wrath of the gods upon themselves and is now being tableslammed across the station. Please stand by.")
+	var/list/areas = list()
+	for(var/area/A in world)
+		if(A.z == SSmapping.station_start)
+			areas += A
+	SEND_SOUND(target, sound('sound/misc/slamofthenorthstar.ogg',volume=40))
+	for(var/area/A in areas)
+		for(var/obj/structure/table/TABLE in A)
+			TABLE.tablepush(target, target)
+			sleep(1)
```
Comment (GoldenAlpharex) [inline code/modules/admin/smites/tabletidestationwide.dm:15]:
```suggestion
	for(var/area/area in world)
		if(area.z == SSmapping.station_start)
			areas += area
	SEND_SOUND(target, sound('sound/misc/slamofthenorthstar.ogg',volume=40))
	for(var/area/area in areas)
		for(var/obj/structure/table/TABLE in area)
			TABLE.tablepush(target, target)
			sleep(1)
```
Single letter variables bad.

**Comment by SmArtKar** [comment]
Make it not break glass tables because else it's punishing the station as well

**Comment by Colovorat** [comment]
Based

File: code/modules/admin/smites/tabletidestationwide.dm (hunk 1 above)
Comment (DuffCreeper) [reply code/modules/admin/smites/tabletidestationwide.dm]: i am the big dum and i am the big sleepy

at least i made table a non single letter *dab

File: code/modules/admin/smites/tabletidestationwide.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+/datum/smite/tabletidestationwide/effect(client/user, mob/living/target)
+	priority_announce("[target] has brought the wrath of the gods upon themselves and is now being tableslammed across the station. Please stand by.")
+	var/list/areas = list()
+	for(var/area/area in world)
+		if(area.z == SSmapping.station_start)
+			areas += area
+	SEND_SOUND(target, sound('sound/misc/slamofthenorthstar.ogg',volume=40))
+	for(var/area/area in areas)
+		for(var/obj/structure/table/TABLE in area)
+			TABLE.tablepush(target, target)
```
Comment (GoldenAlpharex) [inline code/modules/admin/smites/tabletidestationwide.dm:14]:
```suggestion
		for(var/obj/structure/table/table in area)
			table.tablepush(target, target)
```
Variables shouldn't be capitalized, that's for defines.

File: code/modules/admin/smites/tabletidestationwide.dm [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	priority_announce("[target] has brought the wrath of the gods upon themselves and is now being tableslammed across the station. Please stand by.")
+	var/list/areas = list()
+	for(var/area/area in world)
+		if(area.z == SSmapping.station_start)
+			areas += area
+	SEND_SOUND(target, sound('sound/misc/slamofthenorthstar.ogg',volume=40))
+	for(var/area/area in areas)
+		for(var/obj/structure/table/table in area)
+			table.tablepush(target, target)
+			sleep(1)
```
Comment (GoldenAlpharex) [inline code/modules/admin/smites/tabletidestationwide.dm:15]:
```suggestion
	for(var/area/area in areas)
		for(var/obj/structure/table/table in area)
			if(!istype(table, /obj/structure/table/glass)
				table.tablepush(target, target)
				sleep(1)
```
This _should_ make it so it ignores glass tables.

Comment (DuffCreeper) [reply code/modules/admin/smites/tabletidestationwide.dm:15]: im like a kid spoiled on christmas

**Comment by DuffCreeper** [comment]
> Make it not break glass tables because else it's punishing the station as well
With the help of a based individual it now ignores glass at the cost of nuggeting them. The price we pay

I guess one last thing of major concern is the sleep timer but i'm too tired and not exactly versed in the coder way to exactly know a good replacement

**Comment by SuperNovaa41** [comment]
You should make more fixes. Admin bloat + -GBP = angry maintainers

**Comment by ATH1909** [comment]
this is the power... of requiem

**Comment by Cyprex** [comment]
This is just dumb outside of the funny meme value. It's also just very long

**Comment by SuperNovaa41** [comment]
> yay more admin bloat, see #58611

To be fair oranges didn't close it he just fell for the :-1: comment

**Comment by Nobelium-XVIII** [comment]
speedmerge

**Comment by Ryll-Ryll** [comment]
This is a very creative and funny adminsmite

**Comment by Cyberboss** [comment]
Funny but some shit is just too much

**Comment by Wallemations** [comment]
I love the idea of this but I think without some sort of way of choosing how many tables you want them slammed into I doubt it would get used much.
Perforate uses a similar system, I think it would work well.
Also I think making it prioritize tables that are _near_ the victim first would be best, makes it so you don't break a virologist's table on accident.

**Pull request closed without merging**